test/helpers/ERC20PositionHelper.js

```diff
@@ -38,122 +41,218 @@ async function createShortToken(
       }
     ),
     Margin.deployed()
   ]);
 
-  const [bucketLender, ethWrapper, heldToken] = await Promise.all([
-    createBucketLender(openTx, trader),
-    EthWrapperForBucketLender.deployed(),
-    HeldToken.deployed(),
-  ]);
-
+  const
+    [
+      bucketLender,
+      marginToken,
+      owedToken,
+      heldToken
+    ] = await Promise.all([
+      createBucketLender(openTx, trader, deposit, principal),
+      createMarginTokenContract(openTx, type),
+      OwedToken.deployed(),
+      HeldToken.deployed(),
+    ]);
+  openTx.positionOwner = marginToken.address;
 
-  await Promise.all([
-    ethWrapper.depositEth(
-      bucketLender.address,
+  await setupDepositAndPrincipal(
+    accounts,
+    {
+      type,
       trader,
-      {
-        value: new BigNumber('10').times(PRINCIPAL),
-        from: trader
-      },
-    ),
-    issueAndSetAllowance(heldToken, trader, DEPOSIT, TokenProxy.address)
-  ]);
-
+      heldToken,
+      owedToken,
+      bucketLender,
+      deposit,
+      principal,
+    }
+  );
+  // openPositionWithoutCounterparty
   await dydxMargin.openWithoutCounterparty(
     [
-      ERC20ShortFactory.address,
-      WETH9.address,
+      marginToken.address,
+      OwedToken.address,
       HeldToken.address,
-      bucketLender.address
+      bucketLender.address,
     ],
     [
-      PRINCIPAL,
-      DEPOSIT,
-      nonce
+      principal,
+      deposit,
+      nonce,
     ],
     [
       openTx.loanOffering.callTimeLimit,
       openTx.loanOffering.maxDuration,
       openTx.loanOffering.rates.interestRate,
       openTx.loanOffering.rates.interestPeriod,
     ],
-    { from: trader }
+    { from: trader },
   );
-
   return openTx;
 }
 
-async function createBuyOrderForToken(accounts) {
-  const [heldToken, order] = await Promise.all([
-    HeldToken.deployed(),
+async function setupDepositAndPrincipal(
+  accounts,
+  {
+    type,
+    trader,
+    heldToken,
+    owedToken,
+    bucketLender,
+    deposit,
+    principal,
+  }
+) {
+  if (type === POSITION_TYPE.SHORT) {
+    const ethWrapper = await EthWrapperForBucketLender.deployed();
+    await Promise.all([
+      ethWrapper.depositEth(
+        bucketLender.address,
+        accounts[7],
+        {
+          value: new BigNumber('10').times(principal),
+          from: accounts[7],
+        },
+      ),
+      issueAndSetAllowance(heldToken, trader, deposit, TokenProxy.address)
+    ]);
+    return;
+  }
+  if (type === POSITION_TYPE.LONG) {
+    await Promise.all([
+      heldToken.deposit({ value: deposit, from: trader }),
+      heldToken.approve(TokenProxy.address, deposit, { from: trader }),
+      issueAndSetAllowance(
+        owedToken,
+        accounts[7],
+        new BigNumber('10').times(principal),
+        bucketLender.address,
+      ),
+    ]);
+    await bucketLender.deposit(
+      accounts[7],
+      new BigNumber('10').times(principal),
+      { from: accounts[7] },
+    );
+    return;
+  }
```

Comment:
you might want to set this up as a switch statement or an if/else-if/else block where we throw some sort of error when type is not set. And anywhere else where we check type